test/test-functions

```diff
@@ -504,9 +504,8 @@ install_execs() {
     export PKG_CONFIG_PATH=$BUILD_DIR/src/core/
     systemdsystemunitdir=$(pkg-config --variable=systemdsystemunitdir systemd)
     systemduserunitdir=$(pkg-config --variable=systemduserunitdir systemd)
-    egrep -ho '^Exec[^ ]*=[^ ]+' $initdir/{$systemdsystemunitdir,$systemduserunitdir}/*.service \
-         | while read i; do
-         i=${i##Exec*=}; i=${i##[@+\!-]}; i=${i##\!}
+    sed -r -n 's|^Exec[a-zA-Z]*=[@+!-]*([^ ]+).*|\1|gp' $initdir/{$systemdsystemunitdir,$systemduserunitdir}/*.service \
```

Comment:
Not tested, but white spaces may need to be dropped. E.g.,
`  ExecStart    =    !    true`

Reply:
Do we have any of those in tree? This is just for tests, so we don't need to make it completely general.

Reply:
I'm not sure all the prefixes should be dropped silently. In theory, the script should be clever enough to understand that getting "ExecStartPre=-/bin/plymouth" (as in `rescue.service) it shouldn't fail if "plymouth" can't be installed. Though, maybe it's just another bug which hasn't been discovered yet and should probably be fixed in a separate PR.